elide-core/src/main/java/com/yahoo/elide/parsers/state/CollectionTerminalState.java

```diff
@@ -71,8 +71,12 @@ public CollectionTerminalState(Class<?> entityClass, Optional<PersistentResource
         DocumentProcessor includedProcessor = new IncludedProcessor();
         includedProcessor.execute(jsonApiDocument, collection, queryParams);
 
-        DocumentProcessor sortProcessor = new SortProcessor();
-        sortProcessor.execute(jsonApiDocument, collection, queryParams);
+        // if we are using pagination, then we are adding sorting directly into the HQL, etc
+        // todo - is this correct?
+        if (requestScope.getPagination().isDefault()) {
```

Comment:
Yep. For the first pass I just added the notes and the if condition to make sure I would have access to appropriate variables, etc. I will update the sorting today.